test/tests.js

```diff
@@ -29127,3 +29127,141 @@ test("/[a-z]/gim", {
 testFail("/[a-z]/u", "Invalid regular expression flag (1:1)");
 testFail("/[a-z]/y", "Invalid regular expression flag (1:1)");
 testFail("/[a-z]/s", "Invalid regular expression flag (1:1)");
+
+// https://github.com/ternjs/acorn/issues/255
+
+test("(a) = {}", {
```

Comment:
Also, make sure nested patterns like `({ src: ([dest]) } = obj)` do _not_ work (I think that's not being handled yet -- though I didn't test -- and it might be a case for going through `checkPatternErrors` after all)